torch/distributed/_tensor/ops/_matrix_ops.py

```diff
@@ -453,6 +469,27 @@ def scaled_dot_product_efficient_attention_backward_strategy(
     num_heads_dim_sharding.extend([Replicate(), Replicate()])
     single_mesh_dim_strategies.append(num_heads_dim_sharding)
 
+    # Context Parallelism: shards on the sequence dim
+    seq_dim_sharding: PlacementList = [
```

Comment:
ditto: we should not include the strategy here. I think what we should do instead:

since ring attention are currently op handlers, inside the op handlers, we don't need to run sharding propagation, as we know exactly the output sharding is, so we just create a `Shard(2)` output DTensorSpec directly before calling `DTensor._op_dispatcher.wrap(local_results, output_sharding.output_spec)`